gapic/src/main/com/google/gapid/models/AtomStream.java

```diff
@@ -84,6 +86,17 @@ protected void fireLoadEvent() {
     }
   }
 
+  @Override
+  public void onContextSelected(FilteringContext ctx) {
+    if (selection != null && !ctx.contains(selection)) {
+      if (ctx.contains(last(selection))) {
+        selectAtoms(last(selection), 1, false);
+      } else {
+        selectAtoms(ctx.findClosest(selection), false);
```

Comment:
Okay, I guess that's okay assuming the new selection couldn't change context again.